In [1]:
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd 
import statistics
%matplotlib inline
plt.rcParams['figure.figsize'] = (10.0, 5.0)

In [2]:
# Make a blank data frame.
df = pd.DataFrame()

# Populate it with data.

df['Name'] = ['Greg','Marcia','Peter', 'Jan', 'Bobby', 'Cindy', 'Oliver']
df['Age'] = [14, 12, 11, 10, 8, 6, 8]

In [3]:
df

Name  Age
0    Greg   14
1  Marcia   12
2   Peter   11
3     Jan   10
4   Bobby    8
5   Cindy    6
6  Oliver    8

In [4]:
#Calculate mean 
#Using built-in Python functionality.
sum(df['Age']) / len(df['Age'])

9.8571428571428577

In [5]:
#Calculate mean 
# Using NumPy
np.mean(df['Age'])

9.857142857142858

In [6]:
#Calculate median
# Vanilla Python, using the built-in statistics module.

statistics.median(df['Age'])

10

In [7]:
#Calculate median
# Using NumPy.
np.median(df['Age'])

10.0

In [8]:
#Calculate mode
# Return the mode using the statistics module.
statistics.mode(df['Age'])

8

In [9]:
#calculate the Variance 
df['Age'].var()

7.476190476190475

In [10]:
#calculate the the Standard deviation 
np.std(df['Age'], ddof=1)

2.734262327610589

In [11]:
#calculate the the Standard Error
np.std(df['Age'] ,ddof=1) / np.sqrt(len(df['Age']))

1.0334540197243192

Using these estimates, if you had to choose only one estimate of central tendency and one estimate of variance to describe the data, which would you pick and why?  


Ans:  
I would pick the mean 9.8571428571428577 for estimate of Central tendency because that number is a point around which datapoints in a variable cluster.     

  
And the standard deviation 2.734262327610589 for an estimate of variance because for this data it is the best estimate on how much values differ from the central tendency, and how much they differ from each other. 

In [12]:
# Make a blank data frame.
df1 = pd.DataFrame()

# Populate it with data.
#Remove Cousin Oliver and switch with Jessica

df1['Name'] = ['Greg','Marcia','Peter', 'Jan', 'Bobby', 'Cindy', 'Jessica']
df1['Age'] = [14, 12, 11, 10, 8, 6, 1]

In [13]:
df1

Name  Age
0     Greg   14
1   Marcia   12
2    Peter   11
3      Jan   10
4    Bobby    8
5    Cindy    6
6  Jessica    1

In [14]:
#Calculate mean 
#Using built-in Python functionality.
sum(df1['Age']) / len(df1['Age'])

8.8571428571428577

In [15]:
#Calculate mean 
# Using NumPy
np.mean(df1['Age'])

8.857142857142858

In [16]:
#Calculate median
# Vanilla Python, using the built-in statistics module.

statistics.median(df1['Age'])

10

In [17]:
#Calculate median
# Using NumPy.
np.median(df1['Age'])

10.0

In [18]:
#Calculate mode
# Return the mode using the statistics module.
# statistics.mode(df1['Age'])

# Generate a list of unique elements along with how often they occur.
(values, counts) = np.unique(df1['Age'], return_counts=True)

# The location in the values list of the most-frequently-occurring element.
ind = np.argmax(counts)

# The most frequent element.
values[ind]


1

In [19]:
#calculate the Variance 
df1['Age'].var()

18.80952380952381

In [20]:
#calculate the the Standard deviation 
np.std(df1['Age'], ddof=1)

4.3369947901195145

In [21]:
#calculate the the Standard Error
np.std(df1['Age'] ,ddof=1) / np.sqrt(len(df1['Age']))

1.6392299502912862

Nobody likes Cousin Oliver. Maybe the network should have used an even younger actor. Replace Cousin Oliver with 1-year-old Jessica, then recalculate again. Does this change your choice of central tendency or variance estimation methods?

Ans:   
I would pick the median 10.0 for estimate of Central tendency because that number is a point around which datapoints in a variable cluster.    

And the standard deviation 4.3369947901195145 for an estimate of variance because for this data it is the best estimate on how much values differ from the central tendency, and how much they differ from each other. 


In [22]:
# Make a blank data frame.
df2 = pd.DataFrame()

# Populate it with data.

df2['Magazine'] = ['TV Guide','Entertainment Weekly','Pop Culture Today', 'SciPhi Phanatic']
df2['%Fans'] = [20, 23, 17, 5]

In [23]:
df2

Magazine  %Fans
0              TV Guide     20
1  Entertainment Weekly     23
2     Pop Culture Today     17
3       SciPhi Phanatic      5

In [24]:
#Calculate mean 
# Using NumPy
np.mean(df2['%Fans'])

16.25

Based on these numbers, what percentage of adult Americans would you estimate were Brady Bunch fans on the 50th anniversary of the show?

Ans: 
16.25% of the populations were fans of the Brady Bunch.